fondant/compiler.py

```diff
@@ -117,7 +122,17 @@ def _generate_spec(self, pipeline: Pipeline, extra_volumes: list) -> dict:
             command = ["--metadata", json.dumps(asdict(metadata))]
 
             # add in and out manifest paths to command
-            command.extend(["--output_manifest_path", f"{path}/manifest.json"])
+            command.extend(
+                [
+                    "--output_manifest_path",
+                    f"{path}/{safe_component_name}/manifest.json",
+                ]
+            )
+
+            # add component spec to command if does not exist yet
+            if not component_op.arguments.get("component_spec"):
+                component_spec = component_op.component_spec.specification
+                command.extend(["--component_spec", json.dumps(component_spec)])
 
             # add arguments if any to command
             for key, value in component_op.arguments.items():
```

Comment:
could you replace 

```python
for key, value in component_op.arguments.items():
    command.extend([f"--{key}", f"{value}"])
```

with 

```python
for key, value in component_op.arguments.items():
  command.extend([f"--{key}", f"{value}"])
  if isinstance(value, (dict, list)):
      command.extend([f"--{key}", json.dumps(value)])
  else:
      command.extend([f"--{key}", f"{value}"])
```

to fix parsing dictionary, list arguments?

Reply:
done

Reply:
thanks